CarmenPaolaInnesBarron/practica1/phonetics.py

```diff
@@ -0,0 +1,184 @@
+
+Original file is located at
+    https://colab.research.google.com/drive/1uXFwQWkc6c2-D22x3uVvf5SkQO4JKPZ5
+
+# Práctica 1
+"""
+
+import requests as r
+from collections import defaultdict
+import math
+
+lang_codes = {
+  "ar": "Arabic (Modern Standard)",
+  "de": "German",
+  "en_UK": "English (Received Pronunciation)",
+  "en_US": "English (General American)",
+  "eo": "Esperanto",
+  "es_ES": "Spanish (Spain)",
+  "es_MX": "Spanish (Mexico)",
+  "fa": "Persian",
+  "fi": "Finnish",
+  "fr_FR": "French (France)",
+  "fr_QC": "French (Québec)",
+  "is": "Icelandic",
+  "ja": "Japanese",
+  "jam": "Jamaican Creole",
+  "km": "Khmer",
+  "ko": "Korean",
+  "ma": "Malay (Malaysian and Indonesian)",
+  "nb": "Norwegian Bokmål",
+  "nl": "Dutch",
+  "or": "Odia",
+  "ro": "Romanian",
+  "sv": "Swedish",
+  "sw": "Swahili",
+  "tts": "Isan",
+  "vi_C": "Vietnamese (Central)",
+  "vi_N": "Vietnamese (Northern)",
+  "vi_S": "Vietnamese (Southern)",
+  "yue": "Cantonese",
+  "zh": "Mandarin"
+}
+iso_lang_codes = list(lang_codes.keys())
+
+def response_to_dict(ipa_list: list) -> dict:
+    """Parse to dict the list of word-IPA
+
+    Each element of text hae the format:
+    [WORD][TAB][IPA]
+
+    Parameters
+    ----------
+    ipa_list: list
+        List with each row of ipa-dict raw dataset file
+
+    Returns
+    -------
+    dict:
+        A dictionary with the word as key and the phonetic
+        representation as value
+    """
+    result = {}
+    for item in ipa_list:
+        item_list = item.split("\t")
+        result[item_list[0]] = item_list[1]
+    return result
+
+def get_ipa_dict(iso_lang: str) -> dict:
+    """Get ipa-dict file from Github
+
+    Parameters:
+    -----------
+    iso_lang:
+        Language as iso code
+
+    Results:
+    --------
+    dict:
+        Dictionary with words as keys and phonetic representation
+        as values for a given lang code
+    """
+    response = r.get(f"https://raw.githubusercontent.com/open-dict-data/ipa-dict/master/data/{iso_lang}.txt")
+    raw_data = response.text.split("\n")
+    return response_to_dict(raw_data[:-1])
+
+def get_ipa_transcriptions(word: str, dataset: dict) -> list[str]:
+    """Search for word in a given dataset of IPA phonetics
+
+    Given a word this function return the IPA transcriptions
+
+    Parameters:
+    -----------
+    word: str
+        A word to search in the dataset
+    dataset: dict
+        A dataset for a given language code
+    Returns
+      the word phonetic in the given lenguage
+    -------
+    """
+    return dataset.get(word.lower(), "NOT FOUND").split(', ')
+
+def get_similar(word: str, dataset: dict) -> list[str]:
+  """
+    Search for similar words in the given dataset
+    Parameters:
+    -----------
+    word: str
+        A word to search similar words in the dataset
+    dataset: dict
+        A dataset for a given language code
+    Returns
+      the similar words
+    -------
+  """
```

Comment:
Este docstring quedó raro. Con el formato adecuado de [Numpy](https://numpydoc.readthedocs.io/en/latest/format.html) quedaría asi:

```suggestion
  """
    Search for similar words in the given dataset
    
    Parameters
    ----------
    word: str
        A word to search similar words in the dataset
    dataset: dict
        A dataset for a given language code
    Returns
    -----------
    list
      the similar words
  """
```